lib/vtls/mbedtls.c

```diff
@@ -658,20 +616,45 @@ static CURLcode mbed_connect_step1(struct Curl_cfilter *cf,
   }
 
   if(ssl_cert_blob) {
-    /* Unfortunately, mbedtls_x509_crt_parse() requires the data to be null
-       terminated even when provided the exact length, forcing us to waste
-       extra memory here. */
-    unsigned char *newblob = Curl_memdup0(ssl_cert_blob->data,
-                                          ssl_cert_blob->len);
-    if(!newblob)
-      return CURLE_OUT_OF_MEMORY;
-    ret = mbedtls_x509_crt_parse(&backend->clicert, newblob,
-                                 ssl_cert_blob->len + 1);
-    curlx_free(newblob);
+#ifdef MBEDTLS_PEM_PARSE_C
+    /* if DER or a null-terminated PEM just process using
+    * mbedtls_x509_crt_parse().
+    * */
+    if((ssl_cert_type && strcmp(ssl_cert_type, "DER") == 0)
```

Comment:
The cert type is normally checked for case insensitively:

~~~c
if((ssl_cert_type && curl_strequal(ssl_cert_type, "DER"))

Reply:
👍